tree.go

```diff
@@ -417,68 +389,45 @@ var RosterTypeID = network.RegisterMessage(Roster{})
 // adds a UUID which is randomly chosen.
 func NewRoster(ids []*network.ServerIdentity) *Roster {
 	// Don't allow a crash if things are not as expected.
-	if len(ids) < 1 || ids[0].Public == nil {
+	if len(ids) < 1 || ids[0].PublicKey == nil {
 		return nil
 	}
 
-	h := sha256.New()
-	for _, id := range ids {
-		_, err := id.Public.MarshalTo(h)
-		if err != nil {
-			log.Error(err)
-		}
-
-		for _, srvid := range id.ServiceIdentities {
-			_, err = srvid.Public.MarshalTo(h)
-			if err != nil {
-				log.Error(err)
-			}
-		}
-	}
+	// Take a copy of ids, in case the caller tries to change it later.
+	list := make([]*network.ServerIdentity, len(ids))
+	copy(list, ids)
 
-	r := &Roster{
-		ID: RosterID(uuid.NewV5(uuid.NamespaceURL, hex.EncodeToString(h.Sum(nil)))),
-	}
+	return &Roster{List: list}
+}
 
-	// Take a copy of ids, in case the caller tries to change it later.
-	r.List = append(r.List, ids...)
-
-	if len(ids) != 0 {
-		// compute the aggregate key, using the first server's
-		// public key to discover which kyber.Group we should be
-		// using.
-		agg := ids[0].Public.Clone()
-		agg.Null()
-		for _, e := range ids {
-			if e.Public != nil {
-				agg = agg.Add(agg, e.Public)
-			}
-		}
-		r.Aggregate = agg
-	}
-	return r
+// Len returns the length of the roster.
+func (ro *Roster) Len() int {
+	return len(ro.List)
 }
 
 // GetID generates the ID for the list of server identities of the roster
-func (ro *Roster) GetID() (RosterID, error) {
+func (ro *Roster) GetID() RosterID {
 	h := sha256.New()
 	for _, id := range ro.List {
-		_, err := id.Public.MarshalTo(h)
+		_, err := id.PublicKey.WriteTo(h)
 		if err != nil {
-			return RosterID{}, xerrors.Errorf("marshaling: %v", err)
+			// An error at that point would mean something worst is
+			// happening, so we keep the function simple and panic.
```

Comment:
Or, we can have a `RosterBuilder` which populate the `ID` field when building :P